core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/promisehandlers/EpochMarkerHandler.scala

```diff
@@ -2,30 +2,63 @@ package edu.uci.ics.amber.engine.architecture.controller.promisehandlers
 
 import com.twitter.util.Future
 import edu.uci.ics.amber.engine.architecture.controller.ControllerAsyncRPCHandlerInitializer
-import edu.uci.ics.amber.engine.architecture.controller.promisehandlers.EpochMarkerHandler.PropagateEpochMarker
-import edu.uci.ics.amber.engine.architecture.worker.promisehandlers.WorkerEpochMarkerHandler.WorkerPropagateEpochMarker
-import edu.uci.ics.amber.engine.common.ambermessage.EpochMarker
+import edu.uci.ics.amber.engine.architecture.controller.promisehandlers.EpochMarkerHandler.PropagateChannelMarker
+import edu.uci.ics.amber.engine.common.ambermessage.{ChannelID, ChannelMarkerType}
+import edu.uci.ics.amber.engine.common.rpc.AsyncRPCClient.ControlInvocation
 import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.ControlCommand
-import edu.uci.ics.amber.engine.common.virtualidentity.PhysicalOpIdentity
+import edu.uci.ics.amber.engine.common.virtualidentity.{ActorVirtualIdentity, PhysicalOpIdentity}
+import edu.uci.ics.texera.workflow.common.workflow.PhysicalPlan
 
 object EpochMarkerHandler {
 
-  final case class PropagateEpochMarker(destOperator: PhysicalOpIdentity, epochMarker: EpochMarker)
-      extends ControlCommand[Unit]
+  final case class PropagateChannelMarker(
+      sourceOpToStartProp: Set[PhysicalOpIdentity],
+      id: String,
```

Comment:
can we use an object instead of String for the id?

Reply:
done